lib/two_factor_authentication/models/two_factor_authenticatable.rb

```diff
@@ -30,14 +50,15 @@ def authenticate_otp(code, options = {})
           totp.verify_with_drift(code, drift)
         end
 
-        def otp_code(time = Time.now, options = {})
+        def generate_totp_code(time = Time.now, options = {})
+          raise "generate_totp_code called with no otp_secret_key set" if otp_secret_key.nil?
           ROTP::TOTP.new(
-            otp_secret_key,
-            digits: options[:otp_length] || self.class.otp_length
+            otp_secret_key, digits: options[:otp_length] || self.class.otp_length
           ).at(time, true)
         end
 
         def provisioning_uri(account = nil, options = {})
+          raise "provisioning_uri called with no otp_secret_key set" if otp_secret_key.nil?
```

Comment:
I think Rubocop would complain about this in that the first argument should be an error class. I think it's preferable to follow the existing pattern as in the `send_two_factor_authentication_code` method:
```ruby
raise NotImplementedError.new("No default implementation - please define in your class.")
```
Also, I would add more details to the message to explain that `create_otp_secret` needs to be called to set the `otp_secret_key`.